@xen-orchestra/fs/src/abstract.js

```diff
@@ -282,7 +282,8 @@ export default class RemoteHandlerAbstract {
     return entries
   }
 
-  async mkdir(dir: string): Promise<void> {
+  // mutualized to avoid the parallel execution restriction on `mkdir`
+  async _createDir(dir: string): Promise<void> {
```

Comment:
Also, I'm not certain that this is the best fix for this issue, couldn't we simply have remove the limit on `outputFile`?

Because it's a high level method which relies on `mkdir` and `writeFile` which are both already limited?

Reply:
The called `writeFile` method isn't limited because it's a private method. The `outputFile` is indirectly impacted because the issue is localized in the `mktree`. Event if we remove the limit in the `outputFile` it will not be a fix but a work-around.

Reply:
We could update `outputFile` to use `writeFile` instead of `_writeFile`.

IMHO, it makes more sense to have limits on lower level methods, but it might not fit with current architecture.

Reply:
Yes it will work correctly if it use `writeFile` instead of `_writeFile` and it make sens.

Reply:
Done :+1:

Reply:
@julien-f 
> We could update `outputFile` to use `writeFile` instead of `_writeFile`.

Isn't it a bit dangerous since now we have to check that `writeFile` won't trigger a call to a high level method.

> IMHO, it makes more sense to have limits on lower level methods, but it might not fit with current architecture.

Yes, so maybe we should move all the limits to the abstract methods?

@badrAZ `mktree`/`mkdir` will lock the same way. Was that the one you were talking about that also needs to be fixed?

Reply:
@pdonias Yes, but if we considerate that a high level method shouldn't have a lock and should call public methods, it will fix this issue because public methods use the shared lock if it needed.

Reply:
@pdonias I don't know, I think we should go for the simplest fix for this PR.

Eventually, I think limits should not be on high level methods but on implementation methods, because that's why they are for: to not overload backends.